### ISSUE TITLE
Dev dependencies are reported as false positives

### ISSUE THREAD

**Issue by hkdobrev** [maglnet/ComposerRequireChecker#63]
We have dev dependencies like `behat/behat` and `phpstan/phpstan` and then we use them in our code only in local environment. These dependencies are correctly added to `require-dev`, but the checker report them as missing.

| Violations                                                               |
| ------------------------------------------------------------------------ |
| Behat\Behat\Context\Context                                              |
| Clippings\Test\FixturesLoader                                            |
| Behat\Gherkin\Node\TableNode                                             |
| Behat\MinkExtension\Context\RawMinkContext                               |
| Behatch\Context\BaseContext                                              |
| Behatch\HttpCall\HttpCallResultPool                                      |
| Behatch\Json\JsonInspector                                               |
| Behat\Gherkin\Node\PyStringNode                                          |
| Behatch\Json\Json                                                        |
| Behat\Behat\Context\Initializer\ContextInitializer                       |
| Behat\Behat\Context\Argument\ArgumentResolver                            |
| Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException |
| Behat\Testwork\ServiceContainer\Extension                                |
| Behat\Testwork\ServiceContainer\ExtensionManager                         |
| Symfony\Component\DependencyInjection\ContainerBuilder                   |
| Symfony\Component\DependencyInjection\Definition                         |
| Symfony\Component\DependencyInjection\Reference                          |
| Behat\Behat\Context\ServiceContainer\ContextExtension                    |

```
{
    "require-dev": {
        "behat/behat": "^3.4",
        "behat/gherkin": "^4.5",
        "behat/mink": "^1.7",
        "behat/mink-browserkit-driver": "^1.3",
        "behat/mink-goutte-driver": "^1.2",
        "phpstan/phpstan": "^0.8.4",
        "phpstan/phpstan-doctrine": "^0.8",
        "behatch/contexts": "^2.7",
		"treehouselabs/behat-common": "^0.2.1"
 	}
}
```

**Comment by hkdobrev** [comment]
I finally got how the checker decides which files to check. It goes through what Composer would autoload. I had a mix of a Composer-autoloaded modern codebase using more legacy framework code with a custom autoloader.

I was able to remove the custom autoloader using classmap autoloading and now I get more normal violations.
